csrc/ir/utils.cpp

```diff
@@ -755,6 +755,14 @@ std::vector<IterDomain*> allIDsOf(const TensorView* tv) {
       auto all_vals_01 = DependencyCheck::getAllValsBetween(
           {dom0->begin(), dom0->end()}, {dom1->begin(), dom1->end()});
       all_vals.pushBack(all_vals_01);
+      // Getting all vals is not sufficient, because there might be broadcasting
```

Comment:
Does this capture new broadcast IDs if they are just inserted but not used?

Reply:
Good question, let me more thoroughly test it.

Reply:
Added a new test. I do find some issues, so I added a new field `TensorDomain::additional_ids_` to help keep track of all IDs.